cmd/podman/common/sign.go

```diff
@@ -26,9 +27,26 @@ func PrepareSigningPassphrase(pushOpts *entities.ImagePushOptions, signPassphras
 			return err
 		}
 		passphrase = p
-	} else if pushOpts.SignBySigstorePrivateKeyFile != "" {
-		p := ssh.ReadPassphrase()
-		passphrase = string(p)
+	} else {
+		promptForPassphrase := false
+		if pushOpts.SignBySigstorePrivateKeyFile != "" {
+			promptForPassphrase = true
+		}
+		if pushOpts.SignBySigstoreParamFile != "" {
+			// Oops, we need to pre-parse the file because the actual copy implementation is too far from this
```

Comment:
This is _ugly_, and it doesn’t help for Fulcio. Compare https://github.com/containers/common/pull/1288 .

I guess all of creating a `Signer` should happen somewhere around here already?